gax-java/gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -273,42 +276,55 @@ private boolean isDirectPathEnabled() {
     return false;
   }
 
+  private boolean isDirectPathXdsEnabledViaBuilderOption() {
+    return Boolean.TRUE.equals(attemptDirectPathXds);
+  }
+
+  private boolean isDirectPathXdsEnabledViaEnv() {
+    String directPathXdsEnv = envProvider.getenv(DIRECT_PATH_ENV_ENABLE_XDS);
+    return Boolean.parseBoolean(directPathXdsEnv);
+  }
+
   @InternalApi
   public boolean isDirectPathXdsEnabled() {
-    // Method 1: Enable DirectPath xDS by option.
-    if (Boolean.TRUE.equals(attemptDirectPathXds)) {
-      return true;
-    }
-    // Method 2: Enable DirectPath xDS by env.
-    String directPathXdsEnv = envProvider.getenv(DIRECT_PATH_ENV_ENABLE_XDS);
-    boolean isDirectPathXdsEnv = Boolean.parseBoolean(directPathXdsEnv);
-    if (isDirectPathXdsEnv) {
-      return true;
-    }
-    return false;
+    return isDirectPathXdsEnabledViaEnv() || isDirectPathXdsEnabledViaBuilderOption();
```

Comment:
nit: Can you add javadoc for this public method? It is marked as internal, but that unfortunately doesn't stop everyone from using it.

Reply:
Sure, added